app/products/agents/components/agent_post/index.tsx

```diff
@@ -0,0 +1,244 @@
+
+    // Determine reasoning state - use streaming state if available, otherwise use persisted
+    const reasoningSummary = streamingState?.reasoning || persistedReasoning;
+    const isReasoningLoading = streamingState?.isReasoningLoading ?? false;
+    const showReasoning = streamingState?.showReasoning ?? (persistedReasoning !== '');
+
+    // Determine tool calls - use streaming state if available, otherwise use persisted
+    const toolCalls = streamingState?.toolCalls ?? persistedToolCalls;
+
+    // Determine annotations - use streaming state if available, otherwise use persisted
+    const annotations = streamingState?.annotations ?? persistedAnnotations;
+
+    // Check permissions
+    const isRequester = useMemo(() => {
+        return currentUserId ? isPostRequester(post, currentUserId) : false;
+    }, [post, currentUserId]);
+
+    // Determine if generation is in progress (generating or reasoning)
+    const isGenerationInProgress = isGenerating || isReasoningLoading;
+
+    // Show controls based on state and permissions
+    const showStopButton = isGenerationInProgress && isRequester;
+    const hasContent = displayMessage !== '' || reasoningSummary !== '';
+    const showRegenerateButton = !isGenerationInProgress && isRequester && hasContent;
+
+    // Handler for stop button
+    const handleStop = useCallback(async () => {
+        const {error} = await stopGeneration(serverUrl, post.id);
+        if (error) {
+            // Error is already logged in the action
+            // Could show a toast notification here if desired
+        }
+    }, [serverUrl, post.id]);
+
+    // Handler for regenerate button
+    const handleRegenerate = useCallback(async () => {
+        const {error} = await regenerateResponse(serverUrl, post.id);
+        if (error) {
+            // Error is already logged in the action
+            // Could show a toast notification here if desired
+        }
+    }, [serverUrl, post.id]);
+
+    return (
+        <View style={styles.container}>
+            {showReasoning && (
+                <ReasoningDisplay
+                    reasoningSummary={reasoningSummary}
+                    isReasoningLoading={isReasoningLoading}
+                />
+            )}
+            {isPrecontent ? (
+                <View style={styles.precontentContainer}>
+                    <FormattedText
+                        id='agents.generating'
+                        defaultMessage='Generating response...'
+                        style={styles.precontentText}
+                    />
+                    <StreamingIndicator/>
+                </View>
+            ) : (
+                <View style={styles.messageContainer}>
+                    {displayMessage ? (
+                        <Markdown
+                            baseTextStyle={styles.messageText}
+                            value={displayMessage}
+                            theme={theme}
+                            location={Screens.CHANNEL}
```

Comment:
This should be based on the value the post component has, cause this could be channel/thread/search/pinned etc